hbase_sql/src/main/java/com/splicemachine/stream/RemoteQueryClientImpl.java

```diff
@@ -276,4 +278,10 @@ public void close() throws Exception {
         if (olapFuture != null)
             olapFuture.cancel(false);
     }
+
+    @SuppressFBWarnings(value = "RV_RETURN_VALUE_IGNORED", justification = "intended")
+    public Exception getException() throws InterruptedException {
+        olapFutureCallbackInvoked.await(5, TimeUnit.SECONDS);
```

Comment:
Should we handle the timeout? If `await` returns false, we do not have a guarantee that streamListener.getFailure() is going to work as we expect.

Reply:
`streamListener.getFailure()` simply returns the `failure` field. We wait for maximum 5 seconds, hoping that the olapFuture callback is invoked and the `failure` field has the correct exception. But if that's not the case, it's also fine to return whatever exception that has been set to `failure` (the same behavior as before) or `null` if it's not set. We check for `null` on caller side.